### ISSUE TITLE
Request for general guidance for poor predictions

### ISSUE THREAD

**Issue by gfbarron** [keras-team/keras-cv#2522]
Hey, 

I followed the published Keras CV computer vision tutorial (https://keras.io/examples/vision/yolov8/) using the YOLO V8 detector with a custom dataset and have ran into a couple of issues I was hoping to get some help with.  I have tried using Roboflow and the Ultralytics library to fine-tune a model using my dataset and was able to get around 71% mAP so this the primary reason for my post here.  I have looked through issues here and have found some that suggested looking into how data is shuffled etc. and I can confirm that looks fine.

The mAP scores are very low, I was able to get it to improve from 0.001 to 0.06 but I can't seem to get it higher than that even when adding more data and augmentation (roboflow and using the TensorFlow/Keras augmentation pipeline).  

Also, another thing I am noticing is that the model seems to generate a lot of overlapping boxes for the same object. I have tried to play around with NMS parameter values but nothing seems to help. If I set max detections to 10, it generates 10 for the same object with the same score.

Any guidance on how to go about trying to fix this would be greatly appreciated! Let me know if you need any additional information! 

Dataset size 586, 4 classes, training data puts 1 of each class in every image so its evenly balanced. This is enough for now, although I plan to expand to include uneven distribution in images.

This is an example of how I am running inference after training and tweaking NMS:

```python
model.prediction_decoder = keras_cv.layers.NonMaxSuppression(
    bounding_box_format=BOUNDING_BOX_FORMAT,
    from_logits=True,
    iou_threshold=0.5,
    confidence_threshold=0.5,
    max_detections=10
)
```

I am using the PyCOCOCallback during training (as well as others):

```python
    callbacks=[
        keras_cv.callbacks.PyCOCOCallback(  # COCO metrics (AP/AR @ IOU)
            val_ds,
            bounding_box_format=BOUNDING_BOX_FORMAT,
        ),
        keras.callbacks.TensorBoard(  # training progress visualization
            log_dir="training_logs"
        ),
        keras.callbacks.ModelCheckpoint(
            filepath="./model/best_model.keras",
            monitor="val_AP",
            mode="max",  # Save when mAP is maximized
            save_best_only=True,  # Only keep the best model
            verbose=1,
        ),
        keras.callbacks.EarlyStopping(
            patience=20,  # Stop after 25 epochs of no improvement
            monitor="val_AP",
            mode="max",  # Monitor for maximum mAP
            verbose=1,
            restore_best_weights=True,  # Restore weights of the best epoch
        ),
        # VisualizeDetections(),  # visual validation after each epoch
    ],
```

Example of how I am training the model:

```python
backbone = keras_cv.models.YOLOV8Backbone.from_preset(
    "yolo_v8_xs_backbone_coco", load_weights=True
)

prediction_decoder = keras_cv.layers.NonMaxSuppression(
    bounding_box_format=BOUNDING_BOX_FORMAT,
    from_logits=False,
    iou_threshold=0.5,
    confidence_threshold=0.6,
    max_detections=200
)

model = keras_cv.models.YOLOV8Detector(
    num_classes=len(CLASS_IDS),
    bounding_box_format=BOUNDING_BOX_FORMAT,
    backbone=backbone,
    fpn_depth=2,
    prediction_decoder=prediction_decoder,
)

initial_learning_rate = 0.001
lr_schedule = keras.optimizers.schedules.CosineDecay(
    initial_learning_rate,
    decay_steps=1000,
    alpha=0.0
)
optimizer = keras.optimizers.Adam(learning_rate=lr_schedule, global_clipnorm = 10.0)

model.compile(
    classification_loss=keras.losses.CategoricalCrossentropy(),
    box_loss=keras_cv.losses.CIoULoss(bounding_box_format=BOUNDING_BOX_FORMAT),
    optimizer=optimizer,
)
```
